account_invoice_customer_ref_unique/models/account_invoice.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# #############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2010 - 2014 Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+from odoo import api, models, fields, _
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    customer_ref_strip = fields.Char(
+        compute='_compute_customer_ref_strip', store=True)
+
+    @api.multi
+    @api.depends('name')
+    def _compute_customer_ref_strip(self):
+        for invoice in self:
+            invoice_type = invoice.type
+            if invoice_type not in ['out_invoice', 'out_refund']:
```

Comment:
maybe we can remove the previous line as the invoice_type variable is only used once in the if statement